tests/testsuite/path.rs

```diff
@@ -73,44 +72,49 @@ fn cargo_compile_with_nested_deps_shorthand() {
         .build();
 
     p.cargo("build")
-        .with_stderr(
-            "\
+        .with_stderr_data(str![[r#"
 [LOCKING] 3 packages to latest compatible versions
-[COMPILING] baz v0.5.0 ([CWD]/bar/baz)
-[COMPILING] bar v0.5.0 ([CWD]/bar)
-[COMPILING] foo v0.5.0 ([CWD])
-[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
-",
-        )
+[COMPILING] baz v0.5.0 ([ROOT]/foo/bar/baz)
+[COMPILING] bar v0.5.0 ([ROOT]/foo/bar)
+[COMPILING] foo v0.5.0 ([ROOT]/foo)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
         .run();
 
     assert!(p.bin("foo").is_file());
 
-    p.process(&p.bin("foo")).with_stdout("test passed\n").run();
+    p.process(&p.bin("foo"))
+        .with_stdout_data(str![[r#"
+test passed
+
+"#]])
+        .run();
 
     println!("cleaning");
     p.cargo("clean -v")
-        .with_stderr(
-            "[REMOVING] [CWD]/target\n\
-             [REMOVED] [..]",
-        )
+        .with_stderr_data(str![[r#"
+[REMOVING] [ROOT]/foo/target
+[REMOVED] [..] files, [FILE_SIZE]B total
```

Comment:
This highlights the potential benefit of redacting the number of files, as suggested in https://github.com/rust-lang/cargo/pull/14096#discussion_r1644832897.